reminders/script.js

```diff
@@ -1,6 +1,70 @@
+var toDoList = getToDos() || [];
+
+
 function createItem() {
-    var list = document.getElementById("list-items").value;
-    var todo = document.getElementById("todo");
-    todo.insertAdjacentHTML('beforeend', '<input type="checkbox"> ' + list + '<br>');
-    console.log(list)
-}
\ No newline at end of file
+    var newToDo = addNewTodo();
+    updateTodoList(newToDo);
+    saveTodosLocally();
+    addToDoToHTML(newToDo);
+    console.log(toDoList);
+   }
+
+class Todo {
+    constructor (text, completed, id) {
+        this.text = text;
+        this.completed = completed || false;
+        this.id = id || Math.random().toString(36).substring(7);
+    }
+
+}
+
+
+function checkIfCompleted(btn) {
+      for (i = 0; i < toDoList.length; i++) {
+        if (btn.id != toDoList[i].id) {
+            continue;
+        } else if (toDoList[i].id === btn.id && btn.checked === true) {
+            toDoList[i].completed = true;
+        } else {
+            toDoList[i].completed = false;
+        }
+        console.log(toDoList);
+    }
+}
+function addNewTodo() {
+    var input = getInputValue();
+    var newToDo = new Todo(input);
+    return newToDo;
+}
+
+function updateTodoList(newItem) {
+    toDoList.push(newItem);
+}
+
+function getInputValue() {
+    var input = document.getElementById("input").value;
```

Comment:
you don't need to [memoize](https://en.wikipedia.org/wiki/Memoization) here since you're just returning on the next line.